src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -1,81 +1,158 @@
 import { ArrowBackIcon } from '@chakra-ui/icons'
-import { Box, Button, Divider, IconButton, SimpleGrid, Stack, Text } from '@chakra-ui/react'
+import { Box, Button, Divider, IconButton, Link, SimpleGrid, Stack } from '@chakra-ui/react'
+import { chainAdapters, ChainTypes, SwapperType } from '@shapeshiftoss/types'
+import { useState } from 'react'
 import { useFormContext } from 'react-hook-form'
-import { RouterProps } from 'react-router-dom'
+import { useSelector } from 'react-redux'
+import { RouterProps, useLocation } from 'react-router-dom'
 import { Card } from 'components/Card/Card'
 import { HelperTooltip } from 'components/HelperTooltip/HelperTooltip'
 import { Row } from 'components/Row/Row'
 import { SlideTransition } from 'components/SlideTransition'
+import { RawText, Text } from 'components/Text'
+import { useSwapper } from 'components/Trade/hooks/useSwapper/useSwapper'
+import { TradeState } from 'components/Trade/Trade'
+import { useWallet } from 'context/WalletProvider/WalletProvider'
 import { useLocaleFormatter } from 'hooks/useLocaleFormatter/useLocaleFormatter'
 import { bn } from 'lib/bignumber/bignumber'
 import { firstNonZeroDecimal } from 'lib/math'
+import { ReduxState } from 'state/reducer'
 
+import { selectTxHistoryByTxid } from '../helpers'
 import { AssetToAsset } from './AssetToAsset'
 
+type TradeConfirmParams = {
+  fiatRate: string
+}
+
 export const TradeConfirm = ({ history }: RouterProps) => {
-  const { getValues } = useFormContext()
+  const [txid, setTxid] = useState('')
+  const {
+    getValues,
+    handleSubmit,
+    formState: { isSubmitting }
+  } = useFormContext<TradeState<ChainTypes, SwapperType>>()
   const { sellAsset, buyAsset, quote, fees, trade } = getValues()
+  const { executeQuote, reset } = useSwapper()
+  const location = useLocation<TradeConfirmParams>()
+  const { fiatRate } = location.state
   const {
     number: { toFiat }
   } = useLocaleFormatter({ fiatType: 'USD' })
+  const {
+    state: { wallet }
+  } = useWallet()
+  const { chain, tokenId } = sellAsset.currency
+  const asset = tokenId ?? chain
+  const txs = useSelector((state: ReduxState) => {
+    return selectTxHistoryByTxid(state, chain, asset, txid)
+  })
+  const transaction = txs[0]
+
+  const status: chainAdapters.TxStatus = transaction && transaction?.status
+
+  const onSubmit = async () => {
+    if (!wallet) return
+    const result = await executeQuote({ wallet })
```

Comment:
possible `UnhandledPromiseRejection` error

Reply:
This is another ticket https://github.com/shapeshift/lib/issues/152